### PR TITLE
Add ability to mute legend entries by default

### PR DESCRIPTION
**Pull request by philippjfr**
Adds a ``legend_muted`` plot option which mutes the legend entries by default.

- [x] Implements https://github.com/ioam/holoviews/issues/2742

### CODE REVIEW

File: holoviews/plotting/bokeh/element.py [hunk 1]
```diff
@@ -724,7 +724,7 @@ def initialize_plot(self, ranges=None, plot=None, plots=None, source=None):
         else:
             self.handles['xaxis'] = plot.xaxis[0]
             self.handles['x_range'] = plot.x_range
-            self.handles['y_axis'] = plot.yaxis[0]
+            self.handles['yaxis'] = plot.yaxis[0]
```
Comment (philippjfr) [inline holoviews/plotting/bokeh/element.py]: Minor unrelated fix that snuck in.

**Comment by jlstevens** [comment]
Looks good. Merging.
